python/rmm/_lib/device_buffer.pyx

```diff
@@ -0,0 +1,49 @@
+import functools
+import operator
+
+from libc.stdint cimport uintptr_t
+
+import numpy as np
+
+
+cdef class DeviceBuffer:
+
+    def __cinit__(self, ptr=None, size=None):
+        if size is None:
+            size = 0
+
+        cdef void * data
+        if ptr is None:
+            self.c_obj = new device_buffer(<size_t>size)
+        else:
+            data = <void *> <uintptr_t> ptr
+            self.c_obj = new device_buffer(data, size)
+        
+    @property
+    def ptr(self):
+        return int(<uintptr_t>self.c_obj[0].data())
+
+    @property
+    def size(self):
+        return self.c_obj[0].size()
+
+    @staticmethod
+    cdef DeviceBuffer from_ptr(device_buffer *ptr):
```

Comment:
In looking at this API I was expecting it to take a raw pointer as opposed to a pointer to a `device_buffer`. Think we should change the API? Something like `cdef DeviceBuffer from_device_buffer_ptr(...)`?